public_asset/lib/data/network/public_asset_api.dart

```diff
@@ -0,0 +1,129 @@
+import 'package:jmap_dart_client/jmap/mail/extensions/public_asset/get/get_public_asset_response.dart';
+import 'package:jmap_dart_client/jmap/mail/extensions/public_asset/public_asset.dart';
+import 'package:jmap_dart_client/jmap/mail/extensions/public_asset/set/set_public_asset_method.dart';
+import 'package:jmap_dart_client/jmap/mail/extensions/public_asset/set/set_public_asset_response.dart';
+import 'package:public_asset/domain/exceptions/public_asset_exceptions.dart';
+import 'package:tmail_ui_user/main/error/capability_validator.dart';
+import 'package:uuid/uuid.dart';
+
+class PublicAssetApi {
+  const PublicAssetApi(this._httpClient, this._uuid);
+
+  final HttpClient _httpClient;
+  final Uuid _uuid;
+
+  void _checkForSupportedCapability(Session session, AccountId accountId) {
+    if (!CapabilityIdentifier.jmapPublicAsset.isSupported(session, accountId)) {
+      throw const UnsupportedCapabilityException();
+    }
+  }
+
+  MapEntry<Id, PatchObject> _toPatchObjectMapEntry(PublicAsset publicAsset) {
+    final patchObject = PatchObject({
+      PatchObject.identityIdsProperty: const PublicAssetIdentitiesConverter()
+        .toJson(publicAsset.identityIds!),
+    });
+
+    return MapEntry(publicAsset.id!, patchObject);
+  }
+
+  Future<List<PublicAsset>> getPublicAssets(
+    Session session,
+    AccountId accountId,
+    {required List<Id> publicAssetIds}
+  ) async {
+    _checkForSupportedCapability(session, accountId);
+    final requestBuilder = JmapRequestBuilder(_httpClient, ProcessingInvocation());
+    final method = GetPublicAssetMethod(accountId);
+    method.addIds(publicAssetIds.toSet());
+    final invocation = requestBuilder.invocation(method);
+    final response = await (requestBuilder..usings(method.requiredCapabilities))
+      .build()
+      .execute();
+    final publicAssets = response.parse<GetPublicAssetResponse>(
+        invocation.methodCallId,
+        GetPublicAssetResponse.deserialize
+    )?.list ?? [];
+
+    return publicAssets;
+  }
+
+  Future<PublicAsset> createPublicAsset(
+    Session session,
+    AccountId accountId,
+    {
+      required Id blobId,
+      required IdentityId? identityId
+    }
+  ) async {
+    _checkForSupportedCapability(session, accountId);
+    final requestBuilder = JmapRequestBuilder(_httpClient, ProcessingInvocation());
+    final method = SetPublicAssetMethod(accountId);
+    final generateCreateId = Id(_uuid.v1());
+    method.addCreate(
+      generateCreateId,
+      PublicAsset(
+        blobId: blobId,
+        identityIds: identityId != null ? {identityId: true} : {}
+      )
+    );
+    final invocation = requestBuilder.invocation(method);
+    final response = await (requestBuilder..usings(method.requiredCapabilities))
+      .build()
+      .execute();
+    final publicAsset = response.parse<SetPublicAssetResponse>(
+        invocation.methodCallId,
+        SetPublicAssetResponse.deserialize
+    )?.created?[generateCreateId];
```

Comment:
are you sure back-end will response the same id when we create a new public asset?

Reply:
Yes they are the same